Makefile

```diff
@@ -73,55 +63,81 @@ defaults.conf: stbt.conf .stbt-prefix
+    _stbt/xorg.conf.in \
+    stbt-run \
+    stbt-record \
+    stbt-batch \
+    stbt-batch.d/run.py \
+    stbt-batch.d/run-one \
+    stbt-batch.d/report \
+    stbt-batch.d/instaweb \
+    stbt-batch.d/report.py \
+    stbt-batch.d/static/edit-testrun.js \
+    stbt-batch.d/templates/directory-index.html \
+    stbt-batch.d/templates/index.html \
+    stbt-batch.d/templates/testrun.html \
+    stbt-config \
+    stbt-control \
+    stbt-lint \
+    stbt-power \
+    stbt-screenshot \
+    stbt-templatematch \
+    stbt-tv \
+    stbt/__init__.py
+
+INSTALL_VSTB_FILES = \
+    stbt-virtual-stb.d/virtual_stb.py
+
+install: install-core install-virtual-stb
+install-core: stbt.sh defaults.conf $(INSTALL_CORE_FILES)
 	$(INSTALL) -m 0755 -d \
 	    $(DESTDIR)$(bindir) \
-	    $(DESTDIR)$(libexecdir)/stbt \
-	    $(DESTDIR)$(libexecdir)/stbt/_stbt \
-	    $(DESTDIR)$(libexecdir)/stbt/stbt \
-	    $(DESTDIR)$(libexecdir)/stbt/stbt-batch.d \
-	    $(DESTDIR)$(libexecdir)/stbt/stbt-batch.d/static \
-	    $(DESTDIR)$(libexecdir)/stbt/stbt-batch.d/templates \
 	    $(DESTDIR)$(sysconfdir)/stbt \
-	    $(DESTDIR)$(sysconfdir)/bash_completion.d
+	    $(DESTDIR)$(sysconfdir)/bash_completion.d \
+	    $(patsubst %,$(DESTDIR)$(libexecdir)/stbt/%,$(sort $(dir $(INSTALL_CORE_FILES))))
 	$(INSTALL) -m 0755 stbt.sh $(DESTDIR)$(bindir)/stbt
 	$(INSTALL) -m 0755 irnetbox-proxy $(DESTDIR)$(bindir)
-	$(INSTALL) -m 0755 $(tools) $(DESTDIR)$(libexecdir)/stbt
-	$(INSTALL) -m 0644 \
-	    _stbt/__init__.py \
-	    _stbt/config.py \
-	    _stbt/control.py \
-	    _stbt/core.py \
-	    _stbt/gst_hacks.py \
-	    _stbt/irnetbox.py \
-	    _stbt/logging.py \
-	    _stbt/power.py \
-	    _stbt/pylint_plugin.py \
-	    _stbt/state_watch.py \
-	    _stbt/stbt-power.sh \
-	    _stbt/utils.py \
-	    $(DESTDIR)$(libexecdir)/stbt/_stbt
-	$(INSTALL) -m 0644 stbt/__init__.py $(DESTDIR)$(libexecdir)/stbt/stbt
 	$(INSTALL) -m 0644 defaults.conf $(DESTDIR)$(libexecdir)/stbt/stbt.conf
-	$(INSTALL) -m 0755 \
-	    stbt-batch.d/run.py \
-	    stbt-batch.d/run-one \
-	    stbt-batch.d/report \
-	    stbt-batch.d/instaweb \
-	    $(DESTDIR)$(libexecdir)/stbt/stbt-batch.d
-	$(INSTALL) -m 0644 stbt-batch.d/report.py \
-	    $(DESTDIR)$(libexecdir)/stbt/stbt-batch.d
-	$(INSTALL) -m 0644 stbt-batch.d/static/edit-testrun.js \
-	    $(DESTDIR)$(libexecdir)/stbt/stbt-batch.d/static
-	$(INSTALL) -m 0644 \
-	    stbt-batch.d/templates/directory-index.html \
-	    stbt-batch.d/templates/index.html \
-	    stbt-batch.d/templates/testrun.html \
-	    $(DESTDIR)$(libexecdir)/stbt/stbt-batch.d/templates
 	$(INSTALL) -m 0644 stbt.conf $(DESTDIR)$(sysconfdir)/stbt
 	$(INSTALL) -m 0644 stbt-completion \
 	    $(DESTDIR)$(sysconfdir)/bash_completion.d/stbt
+	for filename in $(INSTALL_CORE_FILES); \
+	do \
+	    if [ -x "$$filename" ]; then \
+	        mode=0755; \
+	    else \
+	        mode=0644; \
+	    fi; \
+	    $(INSTALL) -m $$mode $$filename $(DESTDIR)$(libexecdir)/stbt/$$filename; \
+	done
```

Comment:
This (and the rule below) is quite verbose vertically, the mode calculation doesn't deserve quite so much prominence. How about:

```
for filename in $(INSTALL_CORE_FILES); do \
    [[ -x "$$filename" ]] && mode=0755 || mode=0644; \
    $(INSTALL) -m $$mode $$filename $(DESTDIR)$(libexecdir)/stbt/$$filename; \
done
```